src/crypto/openssl/x509_time.h

```diff
@@ -36,7 +36,8 @@ namespace crypto::OpenSSL
   {
     tm tm_time;
     CHECK1(ASN1_TIME_to_tm(time, &tm_time));
-    return std::mktime(&tm_time);
+    // Adjust for timezone (timezone global defined in time.h)
+    return std::mktime(&tm_time) - timezone;
```

Comment:
That does not seem right? I think we want to take UTC inputs, and it's the validation that should change instead.

Reply:
I wonder whether we shouldn't just keep an `ASN1_TIME` for those times and use `ASN1_TIME_print` to convert to strings to avoid `std::mktime`. There's also `ASN1_UTCTIME_adj` which wouldn't require us to make any timezone adjustments I think.

Reply:
As discussed, going for `timegm()` for now. There must be a way to do this with pure C++ standard library function calls to avoid the conversion to `time_t` required by the `ASN1_TIME_adj` function.